### DISCUSSION TITLE
swf Action script3

### DISCUSSION THREAD

**Discussion by rahimhaji** [ruffle-rs/ruffle#4780] [Q&A]
Dear friends,
i have swf files which contains action script3, i tried ruffles on my chrome browser, it says i works only for actionscript2. When can i get ruffels with actionscript3? if any updates please let me know. Please suggest me how can i run my swf with actionscript3? 

looking for your help and suggestions..

Thanks in advance,
Syed Abdul Rahim

**Comment by jhwgh1968** [comment]
While it is on the roadmap there is no public timeline I am aware of for ActionScript 3 support.

If you are interested in this, I would recommend subscribing to notifications on the tracking issue: #1368

You can also try the open source flash player [Lightspark](https://github.com/lightspark/lightspark), which has better ActionScript 3 support. However, it is not complete either.

Hope this helps!

**Reply by rahimhaji** [reply]
Dear Mr.jhwgh1968,

Greetings! thanks for the reply. i tried lightspark, i have downloaded from https://github.com/lightspark/lightspark/releases link. During adding plugins, it gives error msg as "Not a correct imsmanifest.".

Thanks and Regards,
Syed Abdul Rahim

**Reply by jhwgh1968** [reply]
Unfortunately, I have not used the lightspark plugin. I would suggest you look at GitHub issues over there, and see if someone has the same problem. I know their plugin compatibility is not great.

**Reply by rahimhaji** [reply]
Sure. Thanks Mr.jhwgh**1968**
